### PR TITLE
seperate API tests and run them on different DB

### PR DESCRIPTION
**Pull request by individual-it**
The bug that made #320 necessary was not discovered by the tests, because the tests never run with pqsql.
This PR runs the API tests in a separate job and uses real (mysql,pgsql) databases for that.

~~currently blocked by https://github.com/juliushaertl/nextcloud-docker-dev/issues/142~~

### CODE REVIEW

**Comment by juliushaertl** [comment]
New images have been pushed in https://github.com/juliushaertl/nextcloud-docker-dev/actions/runs/4003588199/jobs/6871824562

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -158,3 +144,110 @@ jobs:
...
+            phpVersionMajor: 8
+            phpVersionMinor: 1
+          - nextcloudVersion: master
+            phpVersionMajor: 7
+          - phpVersionMajor: 7
+            phpVersionMinor: 0
+          - phpVersionMajor: 7
+            phpVersionMinor: 1
+          - phpVersionMajor: 8
+            phpVersionMinor: 4
```
Comment (SwikritiT) [inline .github/workflows/ci.yml:172]: what do thye do?

Comment (SwikritiT) [inline .github/workflows/ci.yml:172]: what do they do?

Comment (individual-it) [reply .github/workflows/ci.yml:172]: the matrix will create all possible combinations, but we don't want to test PHP 7.0, 7.1, 8.4

**Review by SwikritiT — APPROVED** [review]

**Pull request merged**